core/src/main/java/org/apache/carbondata/core/datastore/page/VarLengthColumnPageBase.java

```diff
@@ -79,29 +82,29 @@ public void setDoublePage(double[] doubleData) {
   /**
    * Create a new column page for decimal page
    */
-  static ColumnPage newDecimalColumnPage(byte[] lvEncodedBytes, int scale, int precision)
-      throws MemoryException {
+  static ColumnPage newDecimalColumnPage(TableSpec.ColumnSpec columnSpec, byte[] lvEncodedBytes,
+      int scale, int precision) throws MemoryException {
```

Comment:
since columnSpec is passed can we remove scala and precision from methods and get from columnspec?

Reply:
I will try it

Reply:
I will try, but it is a temporary solution. 
I think the correct way is to make DataType a class instead of enum, and keep precision and scale in that.